src/makeInfinityAdder.js

```diff
@@ -6,7 +6,19 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  return function adder(num) {
+    if (typeof num === 'undefined') {
+      return 0;
+    }
+
+    return function(num2) {
+      if (num2 || num2 === 0) {
```

Comment:
is this check necessary?

Reply:
I think so
we have 2 returns, and we need to check wen to return a func and when we return a value
if u mean zero check - without it solution crashes, cause 0 is falsy, but we need to keep going after it. So i made an exception for it